src/App/Fossa/ReleaseGroup/DeleteRelease.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.ReleaseGroup.DeleteRelease (
+  deleteReleaseMain,
+) where
+
+import App.Fossa.Config.ReleaseGroup.DeleteRelease (DeleteReleaseConfig (..))
+import App.Fossa.ReleaseGroup.Common (retrieveReleaseGroupId, retrieveReleaseGroupRelease)
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics, fatalText)
+import Control.Effect.FossaApiClient (FossaApiClient, deleteReleaseGroupRelease, getReleaseGroupReleases, getReleaseGroups)
+import Control.Effect.Lift (Lift)
+import Control.Monad (when)
+import Data.String.Conversion (ToText (..))
+import Effect.Logger (Logger, logInfo, logStdout)
+import Fossa.API.Types (ReleaseGroupRelease (..))
+
+deleteReleaseMain ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has FossaApiClient sig m
+  ) =>
+  DeleteReleaseConfig ->
+  m ()
+deleteReleaseMain DeleteReleaseConfig{..} = do
+  logInfo "Running FOSSA release-group delete-release"
+
+  releaseGroups <- getReleaseGroups
+  maybeReleaseGroupId <- retrieveReleaseGroupId releaseGroupTitle releaseGroups
+  case maybeReleaseGroupId of
+    Nothing -> fatalText $ "Release group `" <> releaseGroupTitle <> "` not found"
+    Just releaseGroupId -> do
+      let releaseGroupIdText = toText releaseGroupId
+
+      releases <- getReleaseGroupReleases releaseGroupIdText
+      when (length releases <= 1) $ fatalText "You are not permitted to delete a release when there is only one release in your release group"
```

Comment:
[nit] This is insignificant here so not change needed, but `length` is pretty inefficient on lists in Haskell. They're linked lists so `length` has to traverse the whole thing to get a length. You can avoid this by pattern matching.